bin/scaffold.js

```diff
@@ -0,0 +1,223 @@
+            .slice(0, 2)
+            .join('/');
+
+    var wtPath = argv.repo ?
+        argv.params.webtask :
+        argv.params.webtask
+            .split('/')
+            .slice(2)
+            .join('/');
+
+    var url = argv.repo || ('https://github.com/' + ghRepoPath);
+
+    return getCommit(url, 'master')
+        .then(function (commit) {
+            return scaffoldFrom(argv, commit, wtPath)
+        })
+        .then(cleanup)
+        .catch(catchErr);
+
+    function catchErr(e) {
+        if(e.message === '\'.tmp\' exists and is not an empty directory') {
+            cleanup();
+
+            return handleScaffold(argv);
+        }
+
+        console.error(e);
+        cleanup();
+    }
+}
+
+function listScaffolds(commit) {
+    var promise = commit ?
+        commit.getTree() :
+        getCommit(DEFAULT_REPO_URL, 'master')
+            .then(function (commit) {
+                return commit.getTree();
+            })
+
+    return promise 
+        .then(function (tree) {
+            return tree.getEntry(DEFAULT_REPO_SAMPLE_DIR);
+        })
+        .then(function (entry) {
+            return entry.getTree();
+        })
+        .then(function (tree) {
+            var entries = tree.entries()
+                .filter(function (entry) {
+                    return Path.extname(entry.toString()) === '.js';
+                });
+
+            var blobs = entries
+                .map(function (entry) {
+                    return entry.getBlob();
+                });
+
+            var names = entries
+                .map(function (entry) {
+                    return Path.basename(entry.toString(), '.js');
+                });
+
+            return Bluebird.all(blobs)
+                .map(function (blob, index) {
+                    var name = names[index];
+                    var jsdoc = JsdParse(blob.toString());
+
+                    return {
+                        name: name,
+                        description: jsdoc[0] ? jsdoc[0].description : 'no description.'
+                    };
+                });
+        })
+        .then(function (scaffolds) {
+            console.log('Available webtasks:'.green.bold);
+
+            scaffolds
+                .forEach(function (scaffold) {
+                    console.log(scaffold.name.bold.white + ':', scaffold.description.grey);
+                });
+        });
+}
+
+function scaffoldFrom(argv, commit, path) {
+    var filename = argv.params.name ? (argv.params.name + '.js') : 'webtask.js';
+
+    return commit
+        .getEntry(path)
+        .then(function (entry) {
+            return entry.getBlob();
+        })
+        .then(function (blob) {
+            return WriteFile(filename, blob.toString())
+                .then(function () {
+                    return blob;
+                });
+        })
+        .then(function (blob) {
+            return GetTaskCfg(argv, blob.toString());
+        })
+        .then(function (cfg) {
```

Comment:
Instead of executing in a child process, check out how I did this for `wt schedule` by delegating to `wt create` (via `Create`): https://github.com/auth0/wt-cli/blob/master/bin/cron.js#L153